src/js/va-letters/containers/VALettersApp.jsx

```diff
@@ -1,19 +1,60 @@
+import PropTypes from 'prop-types';
 import React from 'react';
 import { connect } from 'react-redux';
 
+import RequiredLoginView from '../../common/components/RequiredLoginView';
+
+// This needs to be a React component for RequiredLoginView to pass down
+// the isDataAvailable prop, which is only passed on failure.
+function AppContent({ children, isDataAvailable }) {
+  const unregistered = isDataAvailable === false;
+  let view;
+
+  if (unregistered) {
+    view = (
+      <h4>
+        To view and download your VA letters, you need something something something.
+      </h4>
+    );
+  } else {
+    view = children;
+  }
+
+  return (
+    // TODO: add HTML and CSS when design has settled
+    <div className="row">
+      {view}
+    </div>
+  );
+}
+
 class VALettersApp extends React.Component {
   render() {
-    const { children } = this.props;
     return (
-      <div>{children}</div>
+      <RequiredLoginView
+          authRequired={3}
+          serviceRequired={"evss-letters"}
+          userProfile={this.props.profile}
+          loginUrl={this.props.loginUrl}
+          verifyUrl={this.props.verifyUrl}>
+        <AppContent>
+          {this.props.children}
```

Comment:
Yeah, I wasn't aware of how `this.props.children` can be used either but found this article helpful: https://facebook.github.io/react/docs/composition-vs-inheritance.html

This is the funky way that all of our LOA3-protected apps show or hide content depending on the user (e.g. see [`MessagingApp`](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/js/messaging/containers/MessagingApp.jsx#L59) or [`DisabilityBenefitsApp`](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/js/disability-benefits/containers/DisabilityBenefitsApp.jsx#L46). Then the logic in https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/js/common/components/RequiredLoginView.jsx#L66 adds a prop to `AppContent` (actually to any child React component of `RequiredLoginView`) indicating whether or not to render the content.